operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -553,6 +559,32 @@ func (dc *CassandraDatacenter) GetSuperuserSecretNamespacedName() types.Namespac
 	}
 }
 
+func (dc *CassandraDatacenter) GetReaperUserSecretNamespacedName() types.NamespacedName {
+	name := dc.Spec.ClusterName + "-reaper"
+	namespace := dc.Namespace
+
+	return types.NamespacedName{Name: name, Namespace: namespace}
+}
+
+func (dc *CassandraDatacenter) IsReaperEnabled() bool {
+	return dc.Spec.Reaper != nil && dc.Spec.Reaper.Enabled && dc.Spec.ServerType == "cassandra"
+}
+
+func (dc *CassandraDatacenter) DeployReaper() bool {
+	return dc.IsReaperEnabled() && dc.Status.ReaperStatus.SchemaInitialized
+}
+
+func (dc *CassandraDatacenter) IsAuthenticationEnabled() (bool, error) {
+	b, err := dc.Spec.Config.MarshalJSON()
+	if err != nil {
+		return false, err
+	}
+
+	s := string(b)
+
+	return strings.Contains(s, "PasswordAuthenticator"), nil
```

Comment:
Hmmm... so this config you have here isn't the full story. This will get handed to the config builder that will actually generate the cassandra.yaml. In the past, the defaults for the config builder have sometimes varied from that of the actual defaults (as we do with the SeedProvider impl). Right now, we use the "AllowAllAuthenticator" default in configbuilder, but that's the sort of thing I could see changing. Is there any reason we cannot just assume authentication is enabled?  I think we always create a superuser with credentials irrespective of the authentication setting.

Reply:
Also, if you look at `json.RawMessage` you'll see it's just a `[]byte`:

```
type RawMessage []byte
```

So, if you are going to cheat and not actually parse the JSON, you can just do:

```
return strings.Contains(string(dc.Spec.Config), "PasswordAuthenticator")
```

Then you don't have to return an error from the function either. If the JSON is ill-formed, it will blow up elsewhere.

Reply:
I agree with assuming Authentication is enabled - because will all of this work if they toggle it on after setup?